common/lib/protocol-definitions/src/test/validate0.1024.0.ts

```diff
@@ -0,0 +1,357 @@
+const currentIAttachment: current.IAttachment = get_old_IAttachment();
+declare function set_old_IAttachment(oldVal: old.IAttachment);
+set_old_IAttachment(currentIAttachment);
+
+declare function get_old_ICreateBlobResponse(): old.ICreateBlobResponse;
+export const currentICreateBlobResponse: current.ICreateBlobResponse = get_old_ICreateBlobResponse();
+/*
+declare function set_old_ICreateBlobResponse(oldVal: old.ICreateBlobResponse);
+set_old_ICreateBlobResponse(currentICreateBlobResponse);
+*/
+
+declare function get_old_ITreeEntry(): old.ITreeEntry;
+const currentITreeEntry: current.ITreeEntry = get_old_ITreeEntry();
+declare function set_old_ITreeEntry(oldVal: old.ITreeEntry);
+set_old_ITreeEntry(currentITreeEntry);
+
+declare function get_old_TreeEntry(): old.TreeEntry;
+const currentTreeEntry: current.TreeEntry = get_old_TreeEntry();
+declare function set_old_TreeEntry(oldVal: old.TreeEntry);
+set_old_TreeEntry(currentTreeEntry);
+
+declare function get_old_ITree(): old.ITree;
+const currentITree: current.ITree = get_old_ITree();
+declare function set_old_ITree(oldVal: old.ITree);
+set_old_ITree(currentITree);
+
+declare function get_old_ISnapshotTree(): old.ISnapshotTree;
+const currentISnapshotTree: current.ISnapshotTree = get_old_ISnapshotTree();
+declare function set_old_ISnapshotTree(oldVal: old.ISnapshotTree);
+set_old_ISnapshotTree(currentISnapshotTree);
+
+declare function get_old_ISnapshotTreeEx(): old.ISnapshotTreeEx;
+const currentISnapshotTreeEx: current.ISnapshotTreeEx = get_old_ISnapshotTreeEx();
+declare function set_old_ISnapshotTreeEx(oldVal: old.ISnapshotTreeEx);
+set_old_ISnapshotTreeEx(currentISnapshotTreeEx);
+
+declare function get_old_IVersion(): old.IVersion;
+const currentIVersion: current.IVersion = get_old_IVersion();
+declare function set_old_IVersion(oldVal: old.IVersion);
+set_old_IVersion(currentIVersion);
+
+/*
+declare function get_old_SummaryObject(): old.SummaryObject;
+const currentSummaryObject: current.SummaryObject = get_old_SummaryObject();
+declare function set_old_SummaryObject(oldVal: old.SummaryObject);
+set_old_SummaryObject(currentSummaryObject);
+
+declare function get_old_SummaryTree(): old.SummaryTree;
+const currentSummaryTree: current.SummaryTree = get_old_SummaryTree();
+declare function set_old_SummaryTree(oldVal: old.SummaryTree);
+set_old_SummaryTree(currentSummaryTree);
+
+declare function get_old_ISummaryAuthor(): old.ISummaryAuthor;
+const currentISummaryAuthor: current.ISummaryAuthor = get_old_ISummaryAuthor();
+declare function set_old_ISummaryAuthor(oldVal: old.ISummaryAuthor);
+set_old_ISummaryAuthor(currentISummaryAuthor);
+*/
+
+declare function get_old_ISummaryCommitter(): old.ISummaryCommitter;
+const currentISummaryCommitter: current.ISummaryCommitter = get_old_ISummaryCommitter();
+declare function set_old_ISummaryCommitter(oldVal: old.ISummaryCommitter);
+set_old_ISummaryCommitter(currentISummaryCommitter);
+
+/*
+declare function get_old_SummaryType(): old.SummaryType;
```

Comment:
const enums are not stable across versions in this version of typescript. in version 4.3 of typescript this works. in general enum are not extenable